### ISSUE TITLE
Q: How can i work with relationship

### ISSUE THREAD

**Issue by VictorTomaili** [MoonStorm/Dapper.FastCRUD#9]
How i get complex models or how can i work with virtual model propertiies. There is no documentation about it or i cant find something.

**Comment by MoonStorm** [comment]
Could you be a bit more specific please.

**Comment by VictorTomaili** [comment]
I have n to n relationship. Relation table model has virtual table properties. But how i select they in code? 

```csharp
public partial class RelationTableModel
{
    public virtual int Id {get; set;}
    public virtual int UserId {get; set;}
    public virtual int ProductId {get; set;}
    public virtual UserTableModel UserTable {get; set;} //?
    public virtual ProductTableModel ProductTable {get; set;} //?
}
```

**Comment by MoonStorm** [comment]
Right, I get it now. Relationships are not yet supported. 

They will come, but whether they will allow you to use the properties on your entities or just give you the results separately, that's something I'm still debating. There are pros and cons to each method. People might argue they don't want them attached to the entities, as they won't be used in all the queries and will just pollute the entities. Others might like to have a DB entity that exposes these relationships as well.

**Comment by VictorTomaili** [comment]
Its about config. Like settings in ModelGeneratorConfig.tt file. Who want handle relationship, they set to true. Simple is best but easy is better then when no performance issue. Also when i write something for relationship, i will cant see and ignore something about performance.

**Comment by MoonStorm** [comment]
Added to the FAQ section. Still no support for them in the 2.1 release sorry.

**Comment by NikGovorov** [comment]
I believe it's outside of micro orm scope, please consider Entity Framework or NHIbernate for such scenarios.

**Comment by MoonStorm** [comment]
Whilst working with the library I'm finding myself in situations where I do need at least some basic support for joins. The T4 template is already able to detect relationships, which I'm storing in the discovery step, same as all the other properties. To give you an example, I get entities such as:
```
	[Table("Employee")]
	public partial class Employee
	{
		[Key]
		[DatabaseGenerated(DatabaseGeneratedOption.Computed)]
		public virtual Guid EmployeeId { get; set; }
		[DatabaseGenerated(DatabaseGeneratedOption.Computed)]
		public virtual Guid KeyPass { get; set; }
		public virtual long? WorkstationId { get; set; }
		[ForeignKey("WorkstationId")]
		public virtual Workstation Workstation { get; set; }
	}

	[Table("Workstations")]
	public partial class Workstation
	{
		[Key]
		[DatabaseGenerated(DatabaseGeneratedOption.Identity)]
		public virtual long WorkstationId { get; set; }
		public virtual string Name { get; set; }
		[DatabaseGenerated(DatabaseGeneratedOption.Computed)]
		public virtual int AccessLevel { get; set; }
		public virtual int InventoryIndex { get; set; }
		[ForeignKey("WorkstationId")]
		public virtual IEnumerable<Employee> Employees { get; set; }
	}

```
Obviously these can be set at runtime as well.

Normally the parents or the children won't be returned, but we could allow for an opt-in approach in order to get these as well, by having overloads for most of the methods the library is exposing.  

In our case, we could use ``Find<Employee,Workstation>`` to get an ``Employee`` with its assigned ``Workstation`` or ``Find<Workstation, Employee>`` to get a ``Workstation`` with all the employees allowed to use it. 

I picked the standard ``ForeignKey`` attribute to point to the actual property responsible for the relationship on the current entity. Theoretically this attribute, together with the type of the object in the property, should be enough for us to build the proper SQL.

If I can put this in practice, I suppose the only debate would be having to work with entities that normally are partially populated versus clean entities and callbacks for JOIN type ops.

**Comment by VictorTomaili** [comment]
@NikGovorov I dont think so. firstly i dont want abstract my queriies fully from sql. micro orm is best solution for fully management. and sometimes need extra library for basic jobs. i can write or find and use like this one. And stackowerflow already use dapper and have too much relationships. Because dapper get full sql with joins and can return a multimapped complex entity. micro orm is give full controll for sql. not restrict you for sometings like relationships.

**Comment by VictorTomaili** [comment]
try to use tags for milestones :)

**Comment by MoonStorm** [comment]
I like milestones. I used them in other bigger projects and it lets me know if there are any other planned features/issues missing prior to the release.

**Comment by VictorTomaili** [comment]
i mean git tags. I cant see any tags about milestones for this repository.

**Comment by MoonStorm** [comment]
Ah now I understand what you meant. I'll check the MyGet configuration. So far I've been unimpressed with their build services.

**Comment by MoonStorm** [comment]
Released in 2.4.0. Check the docs for more info.
